native-sql-engine/cpp/src/codegen/arrow_compute/ext/array_taker.h

```diff
@@ -379,6 +379,57 @@ class ArrayTaker<DataType, CType, arrow::enable_if_same<DataType, arrow::StringT
   arrow::MemoryPool* pool_;
 };
 
+class NextArrayTaker : public TakerBase {
+ public:
+  NextArrayTaker(arrow::compute::ExecContext* ctx, arrow::MemoryPool* pool,
+                 std::shared_ptr<arrow::DataType> type)
+      : ctx_(ctx), pool_(pool), type_(type) {
+    std::unique_ptr<arrow::ArrayBuilder> array_builder;
+    arrow::MakeBuilder(ctx_->memory_pool(), type_, &array_builder);
+    builder_.reset(array_builder.release());
+  }
+
+  ~NextArrayTaker() {}
+
+  arrow::Status AddArray(const std::shared_ptr<arrow::Array>& arr) override {
+    cached_arr_.push_back(arr);
+    if (!has_null_ && arr->null_count() > 0) has_null_ = true;
+    return arrow::Status::OK();
+  }
+
+  arrow::Status PopArray() override {
+    cached_arr_.pop_back();
+    has_null_ = false;
+    return arrow::Status::OK();
+  }
+
+  arrow::Status ClearArrays() override {
+    cached_arr_.clear();
+    has_null_ = false;
+    return arrow::Status::OK();
+  }
+
+  arrow::Status TakeFromIndices(ArrayItemIndexS* indices_begin, int64_t length,
```

Comment:
very clean API, how about using this for the rest ArrayTaker?

Reply:
Can be used to for other ArrayTaker. Before, discussed with Binwei, he said  that if it was not a performance issue, did not  have to refactor other code in Gazelle-Plugin.